hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -980,12 +981,18 @@ public void createDirectory(final Path path, final FileSystem.Statistics statist
       if (getAbfsConfiguration().getPrefixMode() == PrefixMode.BLOB) {
         ArrayList<Path> keysToCreateAsFolder = new ArrayList<>();
         checkParentChainForFile(path, tracingContext, keysToCreateAsFolder);
+        boolean blobOverwrite = abfsConfiguration.isEnabledBlobMkdirOverwrite();
 
         HashMap<String, String> metadata = new HashMap<>();
         metadata.put(X_MS_META_HDI_ISFOLDER, TRUE);
+        createFile(path, statistics, blobOverwrite,
+                permission, umask, tracingContext, metadata);
+
         for (Path pathToCreate: keysToCreateAsFolder) {
-          createFile(pathToCreate, statistics, false,
-                  permission, umask, tracingContext, metadata);
+          try {
+            createFile(pathToCreate, statistics, blobOverwrite,
```

Comment:
For parents don't we still need to keep it false always??

Reply:
By default this is set to false and it's on the developers if they want to have parallel mkdirs call with overwrite set to true, they can toggle the flag